.github/workflows/ci.yml

```diff
@@ -0,0 +1,340 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+      
+jobs:
+  run-tests:
+    name: Setup and Run Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - name: Set up Environment Variables
+        run: |
+          echo "DREMIO_SOFTWARE_HOST=localhost" >> $GITHUB_ENV
+          echo "DREMIO_SOFTWARE_USERNAME=dremio" >> $GITHUB_ENV
+          echo "DREMIO_SOFTWARE_PASSWORD=dremio123" >> $GITHUB_ENV
+          echo "DREMIO_DATALAKE=dbt_test_source" >> $GITHUB_ENV
+          echo "DREMIO_DATABASE=dbt_test" >> $GITHUB_ENV
+          echo "DBT_TEST_USER_1=dbt_test_user_1" >> $GITHUB_ENV
+          echo "DBT_TEST_USER_2=dbt_test_user_2" >> $GITHUB_ENV
+          echo "DBT_TEST_USER_3=dbt_test_user_3" >> $GITHUB_ENV
+
+      - name: Create Docker Network
+        run: |
+          docker network create ci-network
+
+      - name: Start MinIO Service
+        run: |
+          docker run -d \
+            --network ci-network \
+            --name minio \
+            -p 9000:9000 \
+            -p 9001:9001 \
+            -e "MINIO_ROOT_USER=admin" \
+            -e "MINIO_ROOT_PASSWORD=password" \
+            minio/minio server /data --console-address ":9001"
+
+      - name: Start Dremio Service
+        run: |
+          docker run -d \
+            --network ci-network \
+            --name dremio \
+            -p 31010:31010 \
+            -p 9047:9047 \
+            -e "DREMIO_JAVA_SERVER_EXTRA_OPTS=-Ddebug.addDefaultUser=true" \
+            dremio/dremio-oss
+
+      - name: Install MinIO Client (mc)
+        run: |
+          curl -O https://dl.min.io/client/mc/release/linux-amd64/mc
+          chmod +x mc
+          sudo mv mc /usr/local/bin/
+
+      - name: Create MinIO bucket
+        run: |
+          for i in {1..12}; do
+            if docker exec minio curl -s http://localhost:9000/minio/health/live; then
+              echo "MinIO is up."
+              break
+            fi
+            echo "Waiting for MinIO to become ready..."
+            sleep 5
+          done
+          if ! docker exec minio curl -s http://localhost:9000/minio/health/live; then
+            echo "MinIO did not become ready in time."
+            exit 1
+          fi
```

Comment:
[SUGGESTION]
Can we store these retries and sleep values in some variables (maybe we can use ENV)